packages/core/redux/bitcoin/thunks.js

```diff
@@ -0,0 +1,212 @@
+/**
+ * Copyright 2017–2018, LaborX PTY
+ * Licensed under the AGPL Version 3 license.
+ */
+
+import BigNumber from 'bignumber.js'
+import bitcoin from 'bitcoinjs-lib'
+import { modalsOpen } from '@chronobank/core-dependencies/redux/modals/actions'
+import {
+  ErrorNoticeModel,
+  TransferNoticeModel,
+  TxExecModel,
+} from '../../models'
+import * as BitcoinActions from './actions'
+import * as BitcoinUtils from './utils'
+import { getSelectedNetwork, getSigner } from '../persistAccount/selectors'
+import { describePendingBitcoinTx } from '../../describers'
+import { getToken } from '../tokens/selectors'
+import { pendingEntrySelector } from './selectors'
+import { notify } from '../notifier/actions'
+
+export const executeBitccoinTransaction = ({ tx, options = null }) => async (dispatch, getState) => {
+  const token = getToken(options.symbol)(getState())
+  const prepared = await dispatch(prepareTransaction(tx, options))
+  const entry = BitcoinUtils.createBitcoinTxEntryModel({
+    tx: prepared,
+    blockchain: token.blockchain(),
+  }, options)
+
+  dispatch(BitcoinActions.createTransaction(entry))
+  dispatch(submitTransaction(entry))
+}
+
+export const prepareTransaction = (tx, { feeMultiplier = 1, satPerByte = null, symbol }) => async (dispatch, getState) => {
+  const state = getState()
+  const token = getToken(symbol)(state)
+  const tokenRate = satPerByte ? satPerByte : token.feeRate()
+  const network = getSelectedNetwork()(state)
+  const prepared = await BitcoinUtils.describeBitcoinTransaction(
+    tx.to,
+    tx.value,
+    {
+      from: tx.from,
+      feeRate: new BigNumber(tokenRate).mul(feeMultiplier),
+      blockchain: token.blockchain(),
+      network,
+    })
+
+  return new TxExecModel({
+    from: tx.from,
+    to: tx.to,
+    amount: new BigNumber(tx.value),
+    fee: new BigNumber(prepared.fee),
+    prepared: prepared.tx,
+    inputs: prepared.inputs,
+    outputs: prepared.outputs,
+  })
+}
+
+const submitTransaction = (entry) => async (dispatch, getState) => {
```

Comment:
This method dispatches only action to open Modal. It is better to move it into src/redux/ui/thunks with appropriate arguments